sopel/modules/admin.py

```diff
@@ -247,61 +247,85 @@ def mode(bot, trigger):
     bot.write(('MODE', bot.nick + ' ' + mode))
 
 
-@sopel.module.require_privmsg("This command only works as a private message.")
-@sopel.module.require_admin("This command requires admin privileges.")
-@sopel.module.commands('set')
-@sopel.module.example('.set core.owner Me')
-def set_config(bot, trigger):
-    """See and modify values of Sopel's config object.
-
-    Trigger args:
-        arg1 - section and option, in the form "section.option"
-        arg2 - value
-
-    If there is no section, section will default to "core".
-    If value is None, the option will be deleted.
+def parse_section_option_value(bot, trigger):
+    """Parse trigger for set/unset to get relevant config elements.
+
+    Returns:
+        - False, if invalid command format; let command handle sending usage.
+        - (), if option not found; reply that option is not found.
+        - (section, section_name, static_sec, option, value) otherwise.
+            - value is `None` if omitted from command.
     """
-    # Get section and option from first argument.
     match = trigger.group(3)
     if match is None:
-        bot.reply("Usage: .set section.option value")
-        return
+        return False  # Invalid command
+
+    # Get section and option from first argument.
     arg1 = match.split('.')
     if len(arg1) == 1:
         section_name, option = "core", arg1[0]
     elif len(arg1) == 2:
         section_name, option = arg1
     else:
-        bot.reply("Usage: .set section.option value")
-        return
+        return False  # Invalid command
     section = getattr(bot.config, section_name)
     static_sec = isinstance(section, StaticSection)
 
     if static_sec and not hasattr(section, option):
         bot.say('[{}] section has no option {}.'.format(section_name, option))
-        return
+        return ()  # Option not found.
+
+    if not static_sec and bot.config.parser.has_option(section, option):
```

Comment:
I do  not understand this line. I must be missing something.

If `section` is *not* a `StaticSection`, AND if `section.option` **does exist**, then it's an error case?

Reply:
Another good catch! This exact line has been around for years, so I didn't give it a proper thought :hushed: 

After dedicating the required brain-power, this line has been corrected (in two ways):
- `... and`**`not`**`bot.config...`
- `...has_option(`**`section_name`**`, ...)`: the `parser.has_option` takes a string as its first argument, but `section` is a `sopel.config.ConfigSection` instance instead

and so this condition was in fact completely borked from its early days.